sap_business_document_processing/document_information_extraction_client/dox_api_client.py

```diff
@@ -0,0 +1,608 @@
+        :param company_code: (optional) The company code of a single record
+        :return: A list of enrichment data records. Returns a list with one item when data_id is given
+        """
+        params = {
+            API_FIELD_CLIENT_ID: client_id,
+            API_REQUEST_FIELD_ENRICHMENT_TYPE: str(data_type)
+        }
+        if data_id is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_ID] = data_id
+        if offset is not None:
+            params[API_REQUEST_FIELD_OFFSET] = offset
+        if limit is not None:
+            params[API_REQUEST_FIELD_LIMIT] = limit
+        if subtype is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_SUBTYPE] = str(subtype)
+        if system is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_SYSTEM] = system
+        if company_code is not None:
+            params[API_REQUEST_FIELD_ENRICHMENT_COMPANYCODE] = company_code
+
+        response = self.get(DATA_ENDPOINT, params=params,
+                            log_msg_before=f'Getting enrichment data records for client {client_id}')
+        self.logger.info(f'Successfully got {len(response.json()[API_FIELD_VALUE])} enrichment data records')
+        return response.json()[API_FIELD_VALUE]
+
+    def delete_all_enrichment_data(self, data_type=None):
+        """
+        This endpoint is deleting all master data records for the account
+        :param data_type: (Optional) The type of enrichment data that should be deleted. Can be given as
+        :enum:`DoxDataType` or string. Options are 'businessEntity' or 'employee'
+        :return: The API endpoint response as dictionary
+        """
+        delete_url = DATA_ASYNC_ENDPOINT
+
+        params = {API_REQUEST_FIELD_ENRICHMENT_TYPE: str(data_type)} if data_type else None
+        response = self.delete(delete_url, json={API_FIELD_VALUE: []}, params=params,
+                               log_msg_before=f"Start deleting all{f' {data_type}' if data_type else ''} enrichment data records")
+
+        job_id = response.json()[API_FIELD_ID]
+        response = self._poll_for_url(DATA_ID_ENDPOINT.format(id=job_id),
+                                      get_status=lambda r: r[API_FIELD_VALUE][API_FIELD_STATUS],
+                                      log_msg_after=f"Successfully deleted all{f' {data_type}' if data_type else ''} enrichment data records")
+        return response.json()
+
+    def delete_enrichment_data(self, client_id, data_type, payload: list, subtype=None, delete_async=False):
+        """
+        Deletes the enrichment data records with the given IDs in the payload
+        :param client_id: The client ID for which the enrichment data was created
+        :param data_type: The type of enrichment data that should be deleted. Can be given as :enum:`DoxDataType` or
+        string. Options are 'businessEntity' or 'employee'
+        :param payload: A list of dictionaries with the form: ``{'id':'', 'system':'', 'companyCode':''}``
```

Comment:
Let's give this argument more meaningful name, like `enrichment_entities_ids`

Reply:
Has been renamed to enrichment_records